src/port1.0/portconfigure.tcl

```diff
@@ -513,87 +590,472 @@ proc portconfigure::configure_get_default_compiler {} {
 }
 
-# internal function to choose compiler fallback list based on platform
-proc portconfigure::get_compiler_fallback {} {
-    global xcodeversion macosx_deployment_target default_compilers configure.sdkroot configure.cxx_stdlib os.major
-
-    # Check our override
-    if {[info exists default_compilers]} {
-        return $default_compilers
+# internal function to determine the Fortran compiler
+proc portconfigure::configure_get_fortran_compiler {} {
+    global configure.compiler
+    set compiler [subst ${configure.compiler}]
+    if {[portconfigure::configure_get_compiler fc ${compiler}] ne ""} {
+        return ${compiler}
     }
 
-    # Check for platforms without Xcode
-    if {$xcodeversion eq "none" || $xcodeversion eq ""} {
-        return {cc}
+    set search_list [option compiler.fortran_fallback]
+    foreach compiler $search_list {
+        set allowed yes
+        foreach pattern [option compiler.blacklist] {
+            if {[string match $pattern $compiler]} {
+                set allowed no
+                break
+            }
+        }
+        if {$allowed &&
+            ([file executable [configure_get_compiler fc $compiler]] ||
+             [compiler_is_port $compiler])
+        } then {
+            return $compiler
+        }
     }
+    ui_warn "All Fortran compilers are either blacklisted or unavailable; defaulting to first fallback option"
+    return [lindex [option compiler.fortran_fallback] 0]
+}
 
-    # Legacy cases
-    if {[vercmp $xcodeversion 4.0] < 0} {
-        set canonical_archs [get_canonical_archs]
-        if {[vercmp $xcodeversion 3.2] >= 0} {
-            if {[string match *10.4u* ${configure.sdkroot}]} {
-                return {gcc-4.0}
-            }
-            # No return here. 3.2.x with newer SDKs than 10.4u is handled below.
-        } elseif {[vercmp $xcodeversion 3.0] >= 0} {
-            if {"ppc" in $canonical_archs || "ppc64" in $canonical_archs} {
-                return {gcc-4.2 apple-gcc-4.2 gcc-4.0 macports-gcc-6 macports-gcc-7}
-            } else {
-                return {gcc-4.2 apple-gcc-4.2 gcc-4.0 macports-clang-3.4 macports-clang-3.3}
-            }
+# thread-local storage only works on Mac OS X Lion and above
+# GCC emulates thread-local storage
+proc thread_local_storage_possible {compiler} {
+    global os.major
+    if {${compiler} eq "clang" || ${compiler} eq "xcode"} {
+        if {${os.major} >= 11} {
+            return yes
         } else {
-            # Xcode 2.x (Tiger)
-            if {"ppc" in $canonical_archs || "ppc64" in $canonical_archs} {
-                if {"i386" in $canonical_archs} {
-                    # universal
-                    return {apple-gcc-4.2 gcc-4.0 macports-gcc-6 macports-gcc-7}
-                } else {
-                    # ppc only
-                    return {apple-gcc-4.2 gcc-4.0 gcc-3.3 macports-gcc-6 macports-gcc-7}
-                }
+            return no
+        }
+        return [expr ${os.major} >= 11]
```

Comment:
unreachable

Reply:
Thank you.
This should be fixed in dd01db91c0dbe2fc79b7265e39e0257ded8cd902.